File: fibonacci_descendente.c
#include <stdio.h>

int fibonacci_desc(int n) {
    if (n < 0) return -1;
    if (n == 0) return 0;
    if (n == 1) return 1;

    return fibonacci_desc(n - 1) + fibonacci_desc(n - 2);
}

int main() {
    int n;

    printf("Ingresa un numero entero: ");
    scanf("%d", &n);

    int resultado = fibonacci_desc(n);

    if (resultado == -1)
        printf("Error: el número debe ser no negativo.\n");
    else
        printf("Fibonacci(%d) = %d\n", n, resultado);

    return 0;
}

File: fibonacci_descendente.py
def fibonacci_descendente(n):
    if n < 0:
        return "El número debe ser no negativo"
    elif n == 0:
        return 0
    elif n == 1:
        return 1
    
    return fibonacci_descendente(n - 1) + fibonacci_descendente(n - 2)


def serie_fibonacci_descendente(n):
    return [fibonacci_descendente(i) for i in range(n + 1)]